### DISCUSSION TITLE
For Radio Veronica?

### DISCUSSION THREAD

**Discussion by ghost** [zenodotus280/BBC-Radio-Relay#14] [General]
Hello,

How would I make this work with just a single stream? 
It resides here: http://playerservices.streamtheworld.com/api/livestream-redirect/VERONICAAAC.aac
Stream is GMT+1
I'd like to listen to it in a GMT-7 (Mountain Time) zone.
but the code is confusing the heck out of me.

I got the basics working but modifying the code and variables to make it do my stream is hard.
Sorry for the inconvenience.

**Comment by zenodotus280** [comment]
As you have seen this is hard-coded for BBC streams but creating a stream-agnostic version is something I would like to do. If you only want one station in one time zone then things are somewhat simplified. The first thing you should look at is `start_radio.sh` - at the end there are `generate_xml` and `start_stream` functions. The `generate_xml` function echoes each line of the XML takes 4 parameters. Let's assume the following:
```
generate_xml () {

    streamname=veronica
    name=Veronica
    genre=OptionalGenreHere
    delay=8

# function continues
```

This should create the XML you need for the `start_stream` function. The `start_stream` function takes two parameters:
eg. `startstream $2 7 #mountain`
In your case...
- streamname = veronica
- delay = 8

Therefore you could add the following:
`startstream veronica 8`
And you can see that it would run the equivalent of:
```
startstream () {
    # $1: stream variable name (ie. 'radio_two')
    # $2: hours of delay

    delay=$2
    ((delay*=60)) #convert delay to minutes
    ((delay*=60)) #convert delay to seconds

echo "sleep $delay && echo \"Starting veronica-8\" && ices2 \"$BASE_FOLDER/ices-xml/gen-veronica-8.xml\" &" >> $logfile
    (sleep $delay && echo "`date` Starting veronica-8" >> $logfile && ices2 "$BASE_FOLDER/ices-xml/gen-veronica-8.xml") &
```

The final piece is `resync` in the cron-scripts folder. This is what initiates the `start_radio.sh` script with the three parameters you're interested in. You would need to add a section such as:
```
# Restart veronica

ps axf | grep downloader | grep veronica | awk '{print "kill -9 " $1}' | sh
ps axf | grep restart-service | grep veronica | awk '{print "kill -9 " $1}' | sh
ps axf | grep ffmpeg | grep veronica | awk '{print "kill -9 " $1}' | sh
ps axf | grep ices2 | grep veronica | awk '{print "kill -9 " $1}' | sh

rm $BASE_FOLDER/audio/*veronica*

$BASE_FOLDER/run-scripts/start_radio.sh "Veronica" veronica "Genre Here"
```

In this last snippet "Veronica" is `$name` and `veronica` is `$streamname` as used by the `generate_xml` function mentioned at the top.

---

Another way of approaching this would simply be to replace every mention of:
- "radio_one" with "veronica"
- "BBC Radio 1" with "Veronica"

---

For the correct URL you should look at `downloader.sh`.

Line 50: `ffmpeg -y -loglevel verbose -timeout -1 -i $BBC_STREAM $filename >> $log 2>&1`

Replace the value in line 18 to `http://playerservices.streamtheworld.com/api/livestream-redirect/VERONICAAAC.aac`

I hope this helps but I haven't tried this myself. If you get errors or something isn't working check the logs and see which script is failing and I'll try and nudge you in the right direction. I'd recommend using the `startCustomstream $2 #user-defined` function instead so you only need to wait 30 seconds to see the result!

**Comment by zenodotus280** [comment]
I have a basic working example created. There is a new branch called "agnostic" that can be used for any stream. See this [installer script](https://raw.githubusercontent.com/zenodotus280/BBC-Radio-Relay/agnostic/installer.sh) for reference: it will create a 30 second delay and won't run any other stations.

**Reply by ghost** [reply]
Wow! What excellent above and beyond replies. Thank you so incredibly much for helping out with all of this!

That installer script totally works; I managed to go to the icecast page that's hosted on port 8000 after running the install script and setting up my vars, and could fetch (and play) the .m3u from it.
I changed the delay variable of startCustomstream to 28800 and then rebooted, I guess that means that in 8 hours I should be able to see it running live on that Icecast page again?

Kind regards
Vixus#1337 (Feel free to add my Discord, it may make things a bit easier for me perhaps)

**Reply by zenodotus280** [reply]
Yeah, you've got it. I've added you on Discord - curious to see how it works for you!
